architecture-building-systems/CityEnergyAnalyst
# Fix code to get buildings' main uses from OSM

This PR does a few things:
1. It fixes an issue (#2947) whereby an incorrect column name that led to OSM building categories not being used to assign CEA typologies (i.e., all buildings were imported as 'MULTI_RES')
2. It moves the names of OSM building categories and corresponding CEA use types into cea.datamanagement.constants in order to (a) keep the zone_helper code cleaner, and (b) allow the list of building categories considered in the future to potentially be expanded without needing to change the zone_helper code
3. It expands the number of [OSM building types](https://wiki.openstreetmap.org/wiki/Key:building) considered to include a wider range of possible values
4. It includes [OSM amenity types](https://wiki.openstreetmap.org/wiki/Key:amenity) as another valid source for data on building use type. According to the OSM wiki, if a building was originally built as, say, an industrial building but was later turned into retail, the it would be labeled `building=industrial` and `amenity=retail`, so the amenity type is actually more important than the building type!

As a result of this, the zone_helper now processes the building use types correctly and considers a number of additional use types:
<img width="1439" alt="Screen Shot 2021-03-26 at 18 07 31" src="https://user-images.githubusercontent.com/18548065/112617001-5db34880-8e5f-11eb-9e23-0deb8bbbc112.png">

To test this PR:
1. Create a new scenario in the CEA dashboard and select any zone you want to – as long as not all buildings are residential
2. Check the typology of the buildings and compare it to the `category` label in CEA
In the master branch, you'll see see that non-residential buildings are also labeled as MULTI_RES, but in this branch they will be correctly imported.
